cowbird/api/webhooks/views.py

```diff
@@ -66,14 +66,15 @@ def handler_fct(handler):
         if callback_url:
             # If something bad happens, set the status as erroneous in Magpie
             LOGGER.warning("Exception occurred while dispatching event [%s], "
-                           "calling Magpie callback url : [%s]", event, callback_url)
+                           "calling Magpie callback url : [%s]", event, callback_url, exc_info=dispatch_exc)
             try:
                 requests.head(callback_url, verify=get_ssl_verify(request), timeout=get_timeout(request))
             except requests.exceptions.RequestException as exc:
                 LOGGER.warning("Cannot complete the Magpie callback url request to [%s] : [%s]", callback_url, exc)
         else:
             LOGGER.warning("Exception occurred while dispatching event [%s].", event, exc_info=dispatch_exc)
-        # TODO: return something else than 200
+        return ax.raise_http(HTTPInternalServerError,
+                             detail=s.UserWebhook_POST_InternalServerErrorResponseSchema.description)
```

Comment:
@fmigneault 
Question here. I added a specific http response here in the case of an exception during the dispatch, as was mentioned in the TODO.
I wasn't sure if it was the best here though, I see the 500 exception is received by Magpie afterwards, but I went with the logic that the error occurred on the server (Cowbird)'s side. I think both the callback and the 500 error change the user's status in Magpie, both changing it to the `WebhookError` status, so it's not a problem.

I was wondering if we wanted a response different than 200, but that would not trigger an exception in Magpie. Should it only be the role of the callback to notice Magpie of an error?

Reply:
I think this is valid. We want the `WebhookError` status to be reported if something failed.

Maybe some contextual details about the exception and the webhook even could be added to the JSON body when raising? This could be useful if we update Magpie to provide more detail about the `WebhookError` status at a later time.

Note: The `return` is not relevant here since it raises.